### PR TITLE
Connect search and detail screen to IMDB API. Implement authorization for detail screen.

### PR DESCRIPTION
**Pull request by ChokchaiThanrukprasert**

### CODE REVIEW

File: web-dev/src/components/AuthenticationLock/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * Put this over component that required authentication to automatically
+ * redirect unauthenticated user to login page
+ * with return URL automatically inserted.
+ *
+ * @param children
+ * @returns {JSX.Element}
+ * @constructor
+ */
+const AuthenticationLock = ({children}) => {
```
Comment (CrimpCity) [inline web-dev/src/components/AuthenticationLock/index.js:14]: Is this similar to professor's secure component?

File: web-dev/src/components/DetailScreen/comment-item.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import {useState} from "react";
 import {Link} from "react-router-dom";
+import {useAuth} from "../../contexts/auth-context";
 
 const CommentItem = ({comment = {}}) => {
 
-  const user = "Potter";
+  const {getUserData} = useAuth();
+  const currentUser = getUserData();
```
Comment (CrimpCity) [inline web-dev/src/components/DetailScreen/comment-item.js:8]: Dumb question but have you tested this with the back end?

**Review by CrimpCity — APPROVED** [review]
This looks good to me.

File: web-dev/src/components/AuthenticationLock/index.js (hunk 1 above)
Comment (ChokchaiThanrukprasert) [reply web-dev/src/components/AuthenticationLock/index.js:14]: Yes, very similar but it adds the returnURL automatically for convenient.

File: web-dev/src/components/DetailScreen/comment-item.js (hunk 2 above)
Comment (ChokchaiThanrukprasert) [reply web-dev/src/components/DetailScreen/comment-item.js:8]: Nope it will be test on the next pull request

**Pull request merged**